### ISSUE TITLE
Empty API response when using wrong user credentials on 18.0.0 and 18.0.1 RC1

### ISSUE THREAD

**Issue by bpcurse** [nextcloud/server#19318]
When trying to query the user list through user metadata OCS API on 18.0.0 and 18.0.1 RC1 using a wrong password or username

`curl -u username:password -X GET 'https://cloud.example.com/ocs/v1.php/cloud/users' -H "OCS-APIRequest: true"`

the API response is empty.

It should return 997 as in previous nextcloud versions:
```
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>997</statuscode>
  <message>Current user is not logged in</message>
  <totalitems></totalitems>
  <itemsperpage></itemsperpage>
 </meta>
 <data/>
</ocs>
```

**Comment by kesselb** [comment]
```
curl -u username:password -X GET 'https://cloud.nextcloud.com/ocs/v1.php/cloud/users' -H "OCS-APIRequest: true"
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>997</statuscode>
  <message>Current user is not logged in</message>
  <totalitems></totalitems>
  <itemsperpage></itemsperpage>
 </meta>
 <data/>
</ocs>
```

I'm not able to reproduce. Probably you have another error in your logs.

**Comment by bpcurse** [comment]
The only thing in the logs is the failed login warning
![failed_login](https://user-images.githubusercontent.com/29312856/73940009-eb924400-48ea-11ea-8051-d35fd4fd86ec.png)

First call is the same as you made, second is to our 16.0.7 instance and the third is to 18.0.1 RC1
![ocs api calls](https://user-images.githubusercontent.com/29312856/73939638-3069ab00-48ea-11ea-91a6-9fd3a4f7dc85.png)
API calls using the correct credentials are working fine. I don't understand this behavior :thinking:
Any idea how to debug this further?

18.0.1 RC1 is running php 7.3.11, mariadb 10.4.10 and nginx 1.16.1 on manjaro i3 linux

**Comment by bpcurse** [comment]
cURL verbose output:

`curl -v -u username:password -X GET 'https://xxx.xxx.xxxxxx.de/ocs/v1.php/cloud/users' -H "OCS-APIRequest: true"`
```
Note: Unnecessary use of -X or --request, GET is already inferred.
*   Trying xxx.xxx.xxx.xxx:443...
* TCP_NODELAY set
* Connected to xxx.xxx.xxxxxxx.de (xxx.xxx.xxx.xxx) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=xxx.xxx.xxxxxx.de
*  start date: xxx xx xx:xx:xx 2019 GMT
*  expire date: xxx xx xx:xx:xx 2020 GMT
*  subjectAltName: host "xxx.xxx.xxxxxx.de" matched cert's "xxx.xxx.xxxxxx.de"
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Server auth using Basic with user 'username'
* Using Stream ID: 1 (easy handle 0x55598c5a5820)
> GET /ocs/v1.php/cloud/users HTTP/2
> Host: xxx.xxx.xxxxxx.de
> authorization: Basic xxxxxxxxxxxxxx
> user-agent: curl/7.68.0
> accept: */*
> ocs-apirequest: true
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 401 
< server: nginx
< date: Thu, 06 Feb 2020 17:28:11 GMT
< content-type: text/html; charset=UTF-8
< location: https://xxx.xxx.xxxxxx.de/login
< expires: Thu, 19 Nov 1981 08:52:00 GMT
< cache-control: no-store, no-cache, must-revalidate
< pragma: no-cache
< set-cookie: oc_sessionPassphrase=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx; path=/; secure; HttpOnly
< content-security-policy: default-src 'self'; script-src 'self' 'nonce-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *; object-src 'none'; base-uri 'self';
< set-cookie: __Host-nc_sameSiteCookielax=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=lax
< set-cookie: __Host-nc_sameSiteCookiestrict=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=strict
< set-cookie: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx; path=/; secure; HttpOnly
< 
* Connection #0 to host xxx.xxx.xxxxxx.de left intact
```
This looks somehow suspicious, although I do not know if it is relevant:
`< expires: Thu, 19 Nov 1981 08:52:00 GMT`

**Comment by bpcurse** [comment]
The test server had a problematic update 17->18 that was followed by a fully manual update.
I think something must have gone wrong there. I cannot reproduce it on another (although apache, not nginx) 18.0.1 instance, so I'll probably write it off as some weird individual issue.

Will delete the instance and perform a clean nextcloud install next week using the same webserver setup on the machine in question and close this if I cannot reproduce it any more.

**Comment by bpcurse** [comment]
I did delete and reinstall Nextcloud 18 and the database on the machine, updated the nginx configuration to the current recommendations in the docs and did a full system upgrade, now using php 7.4. The issue persists.

In the meantime I've checked it on another independent system updated from 17.0.2 to 18.0.2 using nginx, too, and it works.

As no one else has reported this I'm closing the issue as some weird individual server problem.
